src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetGoalTest.kt

```diff
@@ -142,5 +149,71 @@ class GetGoalTest : IntegrationTestBase() {
     val actual = response.responseBody.blockFirst()
     assertThat(actual)
       .hasReference(goalReference)
+    // archive note should be null.
+    assertThat(actual).hasArchiveNote(null)
   }
+
+  @Test
+  fun `should return goal given prison number with archive note`() {
+    // Given a goal is created and archived.
+    createGoal(
+      prisonNumber = prisonNumber,
+      createGoalRequest = aValidCreateGoalRequest(),
+    )
+    val actionPlan = getActionPlan(prisonNumber)
+    val goal = actionPlan.goals[0]
+    val goalReference = goal.goalReference
+
+    val reasonOther = "Because it's Monday"
+    val archiveNote = "an archive note"
+    val archiveGoalRequest = aValidArchiveGoalRequest(
+      goalReference = goalReference,
+      ReasonToArchiveGoal.OTHER,
+      reasonOther,
+      note = archiveNote,
+    )
+
+    archiveAGoal(prisonNumber, goalReference, archiveGoalRequest)
+      .expectStatus()
+      .isNoContent()
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber, goalReference)
+      .bearerToken(
+        aValidTokenWithAuthority(
+          GOALS_RO,
+          privateKey = keyPair.private,
+        ),
+      )
+      .exchange()
+      .expectStatus()
+      .isOk
+      .returnResult(GoalResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasReference(goalReference)
+    // archive note should not be null.
+    assertThat(actual).hasArchiveNote(archiveNote)
+  }
+
+  fun archiveAGoal(
```

Comment:
I think we already have this method in `IntegrationTestBase`, line 231 ?
(I was going to suggest putting it there, then other tests can make use of it in the same way you are here, but then I found we already have it 🤣 )

Reply:
gah - didn't see that although I did look.